collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -338,18 +338,18 @@ public void granted() {
                     setupFields(savedInstanceState);
                     loadForm();
 
-                    /**
+                    /*
                      * Since onResume is called after onCreate we check to see if
                      * it was called without the permissions that are required. If so then
                      * we call it.This is especially useful for cases where a user might revoke
                      * permissions to storage and not know the implications it has on the form entry.
                      */
                     if (onResumeWasCalledWithoutPermissions) {
                         onResume();
+                        refreshCurrentView();
```

Comment:
This is only required when the form gets loaded for the first time. From then onwards, only onActivityResult is concerned to generate new widgets

Reply:
Won't `onResumeWasCalledWithoutPermissions` be false here always because `onCreate` gets called before `onResume`? What do you mean by "for the first time"?

Reply:
Perhaps this code could benefit from some restructuring. I don't think it's generally a good idea to call lifecycle methods directly and this means `onResume` ends up being called twice. It would be better for the work that needs to be done in this case to be pulled out into a method. That might make the intent of the code clearer -- I'm no longer sure what this case is for exactly.

Reply:
I totally agree that this class needs refactoring. Infact I have done a lot of refactoring in the PR https://github.com/opendatakit/collect/pull/2535. This branch was only to apply a patch to the problem with least number of changes. But if you think it's a better idea to fix this problem along with code refactor, then I'd be more than happy to do that too